bigbluebutton-html5/imports/ui/components/settings/component.jsx

```diff
@@ -14,29 +14,76 @@ export default class Settings extends React.Component {
   constructor(props) {
     super(props);
     this.submenus = [];
+    this.state = { activeSubmenu: 0, focusSubmenu: 0 };
+
+    const { isPresenter, role } = this.props;
+
+    this.submenus.push(
+      { componentName: AudioMenu, tabIndex: 3,
+        props: { title: 'Audio', prependIconName: 'icon-', icon: 'bbb-audio', }, },
+      { componentName: VideoMenu, tabIndex: 4,
+        props: { title: 'Video', prependIconName: 'icon-', icon: 'bbb-video', }, },
+      { componentName: ApplicationMenu, tabIndex: 5,
+        props: { title: 'Application', prependIconName: 'icon-', icon: 'bbb-application', }, });
+
+    if (isPresenter || role === 'MODERATOR') {
+      this.submenus.push(
+        { componentName: UsersMenu, tabIndex: 6,
+          props: { title: 'Participants', prependIconName: 'icon-', icon: 'bbb-user', }, });
+    }
   }
 
-  componentWillMount() {
-    /* activeSubmenu represents the submenu in the submenus array to be displayed to the user,
-     * initialized to 0
-     */
-    this.setState({ activeSubmenu: 0 });
-    /* focusSubmenu represents the submenu in the submenus array which currently has focus,
-     * initialized to 0
-     */
-    this.setState({ focusSubmenu: 0 });
-    this.submenus.push({ componentName: AudioMenu, tabIndex: 3,
-      props: { title: 'Audio', prependIconName: 'icon-', icon: 'bbb-audio', }, });
-    this.submenus.push({ componentName: VideoMenu, tabIndex: 4,
-      props: { title: 'Video', prependIconName: 'icon-', icon: 'bbb-video', }, });
-    this.submenus.push({ componentName: ApplicationMenu, tabIndex: 5,
+  renderSettingOptions() {
+
+    return (
+      <div className={styles.full} role='presentation'>
+        <div className={styles.settingsMenuLeft}>
+          <ul className={styles.settingsSubmenu} role='menu'>
+            {this.submenus.map((value, index) => (
+              <li key={index} ref={'submenu' + index} role='menuitem' tabIndex={value.tabIndex}
+                onClick={this.handleClickSubmenu.bind(this, index)}
+                onKeyDown={this.handleKeyDown.bind(this)}
+                onFocus={this.handleFocus.bind(this, index)}
+                className={classNames(styles.settingsSubmenuItem,
+                  index == this.state.activeSubmenu ? styles.settingsSubmenuItemActive : null)}>
+                <Icon key={index} prependIconName={value.props.prependIconName}
+                  iconName={value.props.icon} title={value.props.title}/>
+                <span className={styles.settingsSubmenuItemText}>{value.props.title}</span>
+              </li>
+            ))}
+          </ul>
+        </div>
+        <div className={styles.settingsMenuRight} role='presentation'>
+          {this.renderMenu()}
+        </div>
+      </div>
+    );
+  }
+
+componentWillReceiveProps(nextProps) {
+  this.submenus = [];
+
+  this.submenus.push(
```

Comment:
Why do we need this in 2 places? wouldn't it be better if this was inside `render`? this way it wouldn't need to be in 2 different places